src/command-line/start.js

```diff
@@ -9,12 +10,51 @@ program
 	.option("-B, --bind <ip>"   , "bind")
 	.option("    --public"      , "mode")
 	.option("    --private"     , "mode")
+	.option("    --silent"      , "suppress errors about pid file creation")
 	.command("start")
 	.description("Start the server")
 	.action(function() {
 		var users = new ClientManager().getUsers();
 		var config = Helper.getConfig();
 		var mode = config.public;
+
+		if (config.pidFile) {
+			// setup exception handler to cleanup pid file and log errors
+			process.on("uncaughtException", function(e) {
+				console.log("uncaughtException: " + e.stack || e);
+				process.exit(1);
+			});
+
+			// setup signal handlers to cleanly exit process
+			// which allows the pid file to be removed
+			process.on("SIGINT", process.exit.bind(process, 0));
+			process.on("SIGTERM", process.exit.bind(process, 0));
+
+			// use the default pid file name if the configuration
+			// variable is not a string
+			if (typeof config.pidFile !== "string") {
+				config.pidFile = "shout.pid";
+			}
+
+			// get the absolute path of the pid file to use
+			var pidFilePath = path.resolve(Helper.HOME, config.pidFile);
```

Comment:
as you can see here @JocelynDelalande, although paths are relative to `HOME`, I use the resolve function, which respects absolute paths when given.

IE, if you specify `pidFile: 'shout.pid'` vs `pidFile: '/shout.pid'` the first will be resolved to the home path, and the second will be used verbatim